common/djangoapps/student/helpers.py

```diff
@@ -182,6 +182,8 @@ def check_verify_status_by_course(user, course_enrollments):
                 now = datetime.now(UTC)
                 if deadline is not None and deadline > now:
                     days_until_deadline = (deadline - now).days
+                elif status == VERIFY_STATUS_NEED_TO_REVERIFY:
```

Comment:
@ahsan-ul-haq Is this the bug fix you talked about?
Can you explain this bug a bit? If we don't have this condition, what would go wrong?